android/src/main/java/io/sentry/RNSentryModule.java

```diff
@@ -1,34 +1,265 @@
+    }
+
+    @ReactMethod
+    public void activateStacktraceMerging(Promise promise) {
+        logger.info("TODO: implement activateStacktraceMerging");
+        promise.resolve(true);
+    }
+
+    private Breadcrumb.Level breadcrumbLevel(String level) {
+        switch (level) {
+            case "critical":
+                return Breadcrumb.Level.CRITICAL;
+            case "warning":
+                return Breadcrumb.Level.WARNING;
+            case "info":
+                return Breadcrumb.Level.INFO;
+            case "debug":
+                return Breadcrumb.Level.DEBUG;
+            default:
+                return Breadcrumb.Level.ERROR;
+        }
+    }
+
+    private Event.Level eventLevel(String level) {
+        switch (level) {
+            case "fatal":
+                return Event.Level.FATAL;
+            case "warning":
+                return Event.Level.WARNING;
+            case "info":
+                return Event.Level.INFO;
+            case "debug":
+                return Event.Level.DEBUG;
+            default:
+                return Event.Level.ERROR;
+        }
+    }
+
+    private Level logLevel(int level) {
+        switch (level) {
+            case 1:
+                return Level.SEVERE;
+            case 2:
+                return Level.INFO;
+            case 3:
+                return Level.ALL;
+            default:
+                return Level.OFF;
+        }
+    }
+
+    private Map<String, Object> recursivelyDeconstructReadableMap(ReadableMap readableMap) {
+        ReadableMapKeySetIterator iterator = readableMap.keySetIterator();
+        Map<String, Object> deconstructedMap = new HashMap<>();
+        while (iterator.hasNextKey()) {
+            String key = iterator.nextKey();
+            ReadableType type = readableMap.getType(key);
+            switch (type) {
+                case Null:
+                    deconstructedMap.put(key, null);
+                    break;
+                case Boolean:
+                    deconstructedMap.put(key, readableMap.getBoolean(key));
+                    break;
+                case Number:
+                    deconstructedMap.put(key, readableMap.getDouble(key));
+                    break;
+                case String:
+                    deconstructedMap.put(key, readableMap.getString(key));
+                    break;
+                case Map:
+                    deconstructedMap.put(key, recursivelyDeconstructReadableMap(readableMap.getMap(key)));
+                    break;
+                case Array:
+                    deconstructedMap.put(key, recursivelyDeconstructReadableArray(readableMap.getArray(key)));
+                    break;
+                default:
+                    throw new IllegalArgumentException("Could not convert object with key: " + key + ".");
+            }
+
+        }
+        return deconstructedMap;
+    }
+
+    private List<Object> recursivelyDeconstructReadableArray(ReadableArray readableArray) {
```

Comment:
If this can be typed as `ReadableNativeArray`, then you can use [toArrayList](https://github.com/facebook/react-native/blob/976abf87f213eb3a05277a9956b5e582ba951a2c/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableNativeArray.java#L55)